### PR TITLE
Support GITEA_I_AM_BEING_UNSAFE_RUNNING_AS_ROOT env

### PR DESCRIPTION
**Pull request by wolfogre**
I was trying to run unit tests for Gitea on act runner, by using `make test`.

It failed with log:

```
2024/03/14 03:09:26 ...s/setting/setting.go:180:loadRunModeFrom() [F] Gitea is not supposed to be run as root. Sorry. If you need to use privileged TCP ports please instead use setcap and the `cap_net_bind_service` permission
```

So it will be convenient to skip by setting environment, since it's OK to use root user in job containers.

It's not a bug, but I want to backport it to v1.21 since it doesn't break anything.

### CODE REVIEW

File: modules/setting/setting.go [hunk 1]
```diff
@@ -158,9 +158,14 @@ func loadCommonSettingsFrom(cfg ConfigProvider) error {
 func loadRunModeFrom(rootCfg ConfigProvider) {
 	rootSec := rootCfg.Section("")
 	RunUser = rootSec.Key("RUN_USER").MustString(user.CurrentUsername())
+
 	// The following is a purposefully undocumented option. Please do not run Gitea as root. It will only cause future headaches.
 	// Please don't use root as a bandaid to "fix" something that is broken, instead the broken thing should instead be fixed properly.
 	unsafeAllowRunAsRoot := ConfigSectionKeyBool(rootSec, "I_AM_BEING_UNSAFE_RUNNING_AS_ROOT")
+	if !unsafeAllowRunAsRoot && os.Getenv("GITEA_I_AM_BEING_UNSAFE_RUNNING_AS_ROOT") == "true" {
```
Comment (wxiaoguang) [inline modules/setting/setting.go]:
```
unsafeAllowRunAsRoot = unsafeAllowRunAsRoot || ParseBool(os.Getenv("GITEA_I_AM_BEING_UNSAFE_RUNNING_AS_ROOT"))
```

Comment (wolfogre) [reply modules/setting/setting.go]: c6849304da15c6a87a72544c379ac669a59d27ce

**Review by wxiaoguang — APPROVED** [review]
Hmm a little tricky. In my mind it could be:

```
envUnsafeRunAsRoot, _ := strconv.ParseBool(os.Getenv("GITEA_I_AM_BEING_UNSAFE_RUNNING_AS_ROOT"))
unsafeAllowRunAsRoot = unsafeAllowRunAsRoot || envUnsafeRunAsRoot
```

Either is fine.

**Review by lunny — APPROVED** [review]

File: modules/setting/setting.go [hunk 2]
```diff
@@ -158,9 +159,15 @@ func loadCommonSettingsFrom(cfg ConfigProvider) error {
...
 	RunUser = rootSec.Key("RUN_USER").MustString(user.CurrentUsername())
+
 	// The following is a purposefully undocumented option. Please do not run Gitea as root. It will only cause future headaches.
 	// Please don't use root as a bandaid to "fix" something that is broken, instead the broken thing should instead be fixed properly.
 	unsafeAllowRunAsRoot := ConfigSectionKeyBool(rootSec, "I_AM_BEING_UNSAFE_RUNNING_AS_ROOT")
+	unsafeAllowRunAsRoot = unsafeAllowRunAsRoot || func() bool {
+		v, _ := strconv.ParseBool(os.Getenv("GITEA_I_AM_BEING_UNSAFE_RUNNING_AS_ROOT"))
+		return v
+	}()
+
```
Comment (wxiaoguang) [inline modules/setting/setting.go]:
```suggestion
	unsafeAllowRunAsRoot = unsafeAllowRunAsRoot || util.OptionalBoolParse(os.Getenv("GITEA_I_AM_BEING_UNSAFE_RUNNING_AS_ROOT")).Value()
```

**Pull request merged**